File: 0018-4sum/0018-4sum.java
class Solution {
    public List<List<Integer>> fourSum(int[] nums, int target) {
        HashSet<List<Integer>> ans = new HashSet<>();
        long t = (long)target;
        int n=nums.length;
        Arrays.sort(nums);
        for(int i=0; i<n; i++)
        {
            for(int j=i+1; j<n; j++)
            {
                int k=j+1, l=n-1;
                while(k<l)
                {
                    long sum = (long)nums[i] + nums[j] + nums[k] + nums[l];
                    if(sum == t)
                    {
                        List<Integer> list = new ArrayList<>();
                        list.add(nums[i]);
                        list.add(nums[j]);
                        list.add(nums[k]);
                        list.add(nums[l]);
                        ans.add(list);
                        k++;
                        l--;
                    }
                    else if(sum<t)
                        k++;
                    else
                        l--;
                }
            }
        }
        return new ArrayList(ans);
    }
}

File: 0018-4sum/0018-4sum.py
class Solution:
    def fourSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = set()
        n = len(nums)
        nums.sort()

        for i in range(n): 
            for j in range(i+1, n): 
                k=j+1
                l=n-1
                while k<l:
                    sum = nums[i] + nums[j] + nums[k] + nums[l]
                    if sum == target:
                        ans.add((nums[i], nums[j], nums[k], nums[l]))
                        k += 1;
                        l -= 1;
                    
                    elif sum<target:
                        k += 1
                    else:
                        l -= 1
        return list(ans)